Makefile

```diff
@@ -72,3 +76,51 @@ update-kubernetes-proto: proto-tools
 		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/runtime \
 		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/schema/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/schema \
 		--go-plugin_opt=Mk8s.io/apimachinery/pkg/util/intstr/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/instr;
+	@$(MAKE) format
+
+.PHONY: lint
+lint:
+	@for f in $(all_mods); do \
+        (cd $$(dirname $$f); go run $(golangci_lint) run --timeout 5m); \
+	done
+
+.PHONY: format
+format:
+	@go run $(gofumpt) -l -w .
+	@go run $(gosimports) -local sigs.k8s.io/kube-scheduler-wasm-extension/ -w $(shell find . -name '*.go' -type f)
+
+all_examples :=  $(wildcard ./examples/*/go.mod)
+# all_mods are the go modules including examples. Examples should also be
+# formatted, lint checked, etc. even if they are are built with TinyGo.
+all_mods     := ./internal/e2e/go.mod ./scheduler/go.mod ./guest/go.mod ./kubernetes/proto/go.mod $(all_examples)
+
+.PHONY: tidy
+tidy:
+	@for f in $(all_mods); do \
+        (cd $$(dirname $$f); go mod tidy); \
+	done
+
+.PHONY: build
+build:
+	@# We filter out the examples main packages, as nottinygo cannot compiled \
+     # to a normal platform executable.
+	@for f in $(filter-out $(all_examples), $(all_mods)); do \
+        (cd $$(dirname $$f); go build ./...); \
+	done
+
+.PHONY: test-scheduler
+test-scheduler:
+	@for d in ./scheduler ./internal/e2e; do \
+        (cd $$d; go test ./...); \
+	done
+
+.PHONY: check  # Pre-flight check for pull requests
+check:
+	@$(MAKE) tidy
+	@$(MAKE) build
+	@$(MAKE) lint
+	@$(MAKE) format
```

Comment:
Nit, you may move format up so easy formatting misses fail / can be fixed faster

Reply:
good idea